cmd/argocd/commands/cluster.go

```diff
@@ -137,6 +141,7 @@ func NewClusterAddCommand(clientOpts *argocdclient.ClientOptions, pathOpts *clie
 	command.Flags().StringVar(&systemNamespace, "system-namespace", common.DefaultSystemNamespace, "Use different system namespace")
 	command.Flags().StringArrayVar(&namespaces, "namespace", nil, "List of namespaces which are allowed to manage")
 	command.Flags().StringVar(&name, "name", "", "Overwrite the cluster name")
+	command.Flags().Int64Var(&shard, "shard", -1, "Optional cluster shard number")
```

Comment:
I'm not sure that `optional` adds value here. none of the other flags use it.

Reply:
Agree. Removed optional and added more details.